disnake/client.py

```diff
@@ -1568,6 +1569,40 @@ async def fetch_guild(self, guild_id: int, /) -> Guild:
         data = await self.http.get_guild(guild_id)
         return Guild(data=data, state=self._connection)
 
+    async def fetch_guild_preview(
+        self,
+        guild_id: int,
+        /,
+    ) -> GuildPreview:
+        """|coro|
+
+         Retrieves a :class:`.GuildPreview` from the given ID. Your bot does not have to be in this guild.
+
+        .. note::
+
+            This method may fetch any guild that has ``DISCOVERABLE`` in :attr:`Guild.features`,
+            but this information can not be known ahead of time.
+
+            This will work for any guild that you are in.
+
+        Parameters
+        ----------
+        guild_id: :class:`int`
+            The ID of the guild to to retrieve a preview object.
+
+        Raises
+        ------
+        NotFound
+            Retrieving the guild preview failed.
```

Comment:
What about `Forbidden` and `HTTPException`?

Reply:
HTTPException missing is an oversight for a malformed request, but if you do not have access to a guild's preview, its a 404.